scripts/load_testing/locustfile.py

```diff
@@ -0,0 +1,208 @@
+    dom = pyquery.PyQuery(resp.content)
+    try:
+        code = dom.find("#code")
+        # It's prompting for a 2FA code, so we know it was a valid login
+        resp = t.client.post(
+            '/login/two_factor/sms',
+            data = {
+                'code': code.attr('value'),
+                'authenticity_token': authenticity_token(dom),
+                'commit': 'Submit'
+            }
+        )
+        resp.raise_for_status()
+        print('Sign in complete.')
+    except Exception as error:
+        print(error)
+
+
+def logout(t):
+    """
+    Takes a locustTask object and signs you out.
+    Naively assumes the user is actually logged in already.
+    """
+    print('Beginning sign out')    
+    resp = t.client.get('/')
+    resp.raise_for_status()
+    dom = pyquery.PyQuery(resp.content)
+
+    try:
+        sign_out_link = dom.find('a[href="/api/saml/logout"]')
+        # Authentication is now complete.
+        # We've confirmed by the presence of the sign-out link. 
+        # We can now have the person sign out.
+        resp = t.client.get(sign_out_link.attr('href'))
+        resp.raise_for_status()
+        dom = pyquery.PyQuery(resp.content)
+        # Let's confirm:
+        print(dom.find('div.alert-success').eq(0).text())
+    except Exception as error:
+        print(error)
+
+
+def change_pass(t):
+    """
+    Takes a locustTask and naively expects an already logged in person,
+    this navigates to the account (which they should already be on, post-login)
+    """
+    
+    resp = t.client.get('/account')
+    resp.raise_for_status()
+    dom = pyquery.PyQuery(resp.content)
+    edit_link = dom.find('a[href="/manage/password"]')
+
+    try:
+        resp = t.client.get(edit_link.eq(0).attr('href'))
+    except Exception as error:
+        print("""
+            There was a problem finding the edit pass link.
+            Most likely, you're hitting an OTP cap with this user.
+            We are unable to change the password and are exiting.
+            """
+        )
+        print(error)
+        return
+
+    resp.raise_for_status()
+    dom = pyquery.PyQuery(resp.content)
+    # do we want to check that you're on the page where you insert your old pass?
+    # we could look for form action before proceeding to the post?
+    # Note that if you follow the form action it may take you
+    # to /reauthn and we would need to handle that case. 
+    # To keep it simple for now we're skipping reauth
+    if '/manage/password' in resp.url:
+        resp = t.client.post(
+            resp.url,
+            data = {
+                'update_user_password_form[password]': "thisisanewpass",
+                'authenticity_token': authenticity_token(dom),
+                '_method': 'patch',
+                'commit': 'update'
+            }
+        )
+        resp.raise_for_status()
+        dom = pyquery.PyQuery(resp.content)
+        # Now change it back
+        resp = t.client.post(
```

Comment:
@jmhooper Is there anything going on in /manage/password that would have a problem with the switch back to the original password here? I'm getting a 404, similar to when it didn't have _method. Interestingly, the 404 doesn't seem to happen all the time. Any thoughts?

Reply:
Hmmm 🤔 , I'm not sure. A 404 is a weird thing for it to render here. I'd think that means the request is not getting routed correctly, meaning a malformed request url or we're using the wrong HTTP method.

Reply:
What I saw earlier with the _method was that if there's anything about the POST that /manage/password doesn't like, it appears to respond with a 404. 

What's interesting is that code above -- L116-L124 is fine. But here, when I try to change it back to the original value, it chokes. 

Even more interesting, it appears the choking is inconsistent. Like maybe I'm hitting an OTP cap again or something.

Reply:
It 404s if the `_method` param is missing because that is used for routing

Reply:
OK, I get that, and it works correctly above. But there's something about changing the password right back here that it doesn't like.